plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineDirectManifest.kt

```diff
@@ -18,12 +19,14 @@ import java.security.interfaces.RSAPublicKey
 import javax.inject.Inject
 import kotlin.io.path.absolutePathString
 
-open class GenerateGramineManifest @Inject constructor(
-        objects: ObjectFactory,
-        private val buildType: BuildType
+open class GenerateGramineDirectManifest @Inject constructor(
+    objects: ObjectFactory,
+    private val buildType: BuildType
 ) : ConclaveTask() {
     companion object {
-        const val MANIFEST_TEMPLATE = "java.manifest.template"
+        const val MANIFEST_TEMPLATE = "$GRAMINE_MANIFEST.template"
+        const val PYTHON_ENCLAVE_SIZE = "8G"
+        const val JAVA_ENCLAVE_SIZE = "4G"
```

Comment:
Add TODO that the enclave_size property should be the sum of the heap and stack size configs. You can probably link to this ticket: https://r3-cev.atlassian.net/browse/CON-1221

Reply:
Nope, I am quite against this. First of all Gramine `enclave_size` needs to be power of 2, not really sure we can easily sum two numbers and achieve that, if they are not the same power of 2 value!

Reply:
That can be solved by just going up to the nearest power of 2.

Why is this a bad thing? We have these two configs in the enclave build.gradle and they need to be wired up in Gramine. What do you suggest we do instead?

Reply:
Add a TODO for the suggested solution.

Reply:
maxHeapSize and maxStackSize are more Graal parameters than SGX related and they can be quite different.
Rounding up might waste a lot of memory.
I would suggest a Gramine specific parameter, `gramineEnclaveSize`. I feel that at some point we might need two separate sections in the `build.gradle`

Reply:
I don't like the addition of another parameter for the enclave size, maybe re-use the same parameter and warn the user that it's rounded to the next power of two, or produce an error telling them to make it a power of two?

Reply:
Which parameter would you reuse?

Reply:
Ignore my previous comment, was under the impression we had an `enclaveSize` parameter, looks like we don't!

I'd just connect it and force the user to round it up to a power of two, or print a warning that it is being done. Definitely **don't** do something like `(threadCount * stackSize) + heapSize` though.

Reply:
Not sure I understand, would you use `(threadCount * stackSize) + heapSize` and round that up? Or `heapSize` and round that up?
In both cases I see risky situations and waste of almost 2^(k-1) memory.

Reply:
Apologies, I don't understand.
Is there the possibility of having more stacks in the enclave than `sgx.thread_num`? My understanding is no.
What you mean by "normal Posix call"? My understanding is that in Gramine they have reimplemented them.

But if the users set the `heapSize` as power of 2 thinking that it is the actual heap and hence `enclave_size` is power of 2, wouldn't it be misleading/wrong that we don't include stack (and PAL manifest) sizes as well?
And if the users set `heapSize` to 2.1G for example , wouldn't be a waste do round up?
Rounding up `(threadCount * stackSize) + heapSize` makes more sense to me that `heapSize` only, but I honestly dislike both options, as they are confusing and misleading.

Reply:
Yes, you can have as many threads/stacks in the enclave as you like with gramine, irrespective of the value of `sgx.thread_num`. This is what I was talking about yesterday. I'd also wager that gramine does _not_ greedily allocate stack memory at startup. So no, 100% _does not make sense_ to make this a function of stack size.

As for the rounding, I think you're over-thinking this. People rarely plan the memory utilisation of their programs ahead of time. They run them, and if they run out of memory, they allocate more. So long as they are aware that the rounding is happening, or they are forced to round themselves, I'd say that's good enough.

Reply:
>Yes, you can have as many threads/stacks in the enclave as you like with gramine, irrespective of the value of sgx.thread_num. 

Are you sure that also applies to concurrent ECALLs? Because I'm not convinced it is.